File: SchoolProject-angular/src/app/Models/Student.ts
// export class Student{
//   id?: number;
//   name?: string;
//   surname?: string;
//   phone?: number;
// }


export class Student{

  // ! We add a constructor to clean the fields of a previous selected student
  constructor() {
    this.id = 0;
    this.name = '';
    this.surname = '';
    this.phone = 0;
  }
  id?: number;
  name?: string;
  surname?: string;
  phone?: number;
}

File: SchoolProject-WebAPI/Models/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SchoolProject_WebAPI.Models
{
    public class Student
    {

        public Student(){}
        public Student(int id, string name, string surname, string phone)
        {
            this.Id = id;
            this.Name = name;
            this.Surname = surname;
            this.Phone = phone;
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public string Surname { get; set; }
        public string Phone { get; set; }

        public IEnumerable<StudentSubject> StudentsSubjects { get; set; }
    }
}